### ISSUE TITLE
Training with 0 as the anomalous label

### ISSUE THREAD

**Issue by layadas** [liznerski/fcdd#71]
I am trying to understand how I can edit the code so that I can safely use an anomalous label of `0` as is done in the [HSC](https://arxiv.org/pdf/2006.00339.pdf) paper. I went into the loss function definitions in the FCDD [trainer](https://github.com/liznerski/fcdd/blob/master/python/fcdd/training/fcdd.py#L7) and it looks like the loss is hard-coded to consider `0` as nominal and `1` as anomalous. I want to rewrite the loss function so that `0` is anomalous and `1` is nominal. How can I achieve this?

**Comment by liznerski** [comment]
Hi. 

You're right. Apparently this got hard-coded at some point. After a quick check, it seems that the datasets still support a custom nominal label, but the trainer and losses don't. We need to pass the nominal label to the trainer [here](https://github.com/liznerski/fcdd/blob/master/python/fcdd/training/setup.py#L128), [here](https://github.com/liznerski/fcdd/blob/master/python/fcdd/training/super_trainer.py#L15), [here](https://github.com/liznerski/fcdd/blob/master/python/fcdd/training/super_trainer.py#L53), and [here](https://github.com/liznerski/fcdd/blob/master/python/fcdd/training/fcdd.py#L8). Then, the trainer needs to pass the nominal label definition to the loss invocations at all relevant places. For example, the losses are called [here](https://github.com/liznerski/fcdd/blob/master/python/fcdd/training/bases.py#L68), [here](https://github.com/liznerski/fcdd/blob/master/python/fcdd/training/bases.py#L184), [here](https://github.com/liznerski/fcdd/blob/master/python/fcdd/training/bases.py#L190), [here](https://github.com/liznerski/fcdd/blob/master/python/fcdd/training/bases.py#L313),  and [here](https://github.com/liznerski/fcdd/blob/master/python/fcdd/training/bases.py#L321). Then, we need to redefine the loss function to process the nominal label and invert the scores for the AuROC computation. In conclusion, it's perhaps easier to just swap the labels of your dataset. ;)

Btw, there is an updated version of the HSC paper published in TMLR: https://arxiv.org/pdf/2205.11474.pdf. 

Hope it helps!

**Comment by layadas** [comment]
Thank you for your suggestion. It does appear easier to switch the labels. Thank you for the link to the paper.
